### ISSUE TITLE
An error occurs when used with the 'Sliver' widget.

### ISSUE THREAD

**Issue by Patrick386** [fleather-editor/fleather#163]
When used with the 'Sliver' widget, the error occurs as follows:

```
class TextEditorX extends StatefulWidget {
  const TextEditorX({super.key});

  @override
  State<TextEditorX> createState() => _TextEditorXState();
}

class _TextEditorXState extends State<TextEditorX> {
  late ParchmentDocument document;
  late FleatherController _controller ;

  @override
  void initState() {
    document =   ParchmentDocument.fromJson(json);
    _controller  = FleatherController(document);
    super.initState();
  }
  @override
  Widget build(BuildContext context) {

    return SliverList(
      delegate: SliverChildListDelegate([
        FleatherToolbar.basic(controller: _controller),
        FleatherEditor(controller: _controller,
          padding: EdgeInsets.only(
            left: 16,
            right: 16,
            bottom: MediaQuery.of(context).padding.bottom,
          ),
          maxContentWidth: 800.0,
          maxHeight: 300.0,),
        //or
        // FleatherField(controller: _controller)
      ]),

    );
  }
}
```

```
======== Exception caught by widgets library =======================================================
The following assertion was thrown while applying parent data.:
Incorrect use of ParentDataWidget.

The ParentDataWidget Flexible(flex: 1) wants to apply ParentData of type FlexParentData to a RenderObject, which has been set up to accept ParentData of incompatible type WrapParentData.

Usually, this means that the Flexible widget has the wrong ancestor RenderObjectWidget. Typically, Flexible widgets are placed directly inside Flex widgets.
The offending Flexible is currently placed inside a Wrap widget.

The ownership chain for the RenderObject that received the incompatible parent data was:
  Semantics ← Image ← Container ← Flexible ← Wrap ← SizedBox ← Center ← SliverToBoxAdapter ← Viewport ← IgnorePointer-[GlobalKey#e0d11] ← ⋯
When the exception was thrown, this was the stack: 
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 294:49      throw_
```

**Comment by Amir-P** [comment]
I was able to use your widget in a `CustomScrollView` and it works fine. From the logs you've provided I think your issue is not related to `TextEditorX` widget. @Patrick386 

```dart
import 'package:fleather/fleather.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(const FleatherApp());
}

class FleatherApp extends StatelessWidget {
  const FleatherApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) => MaterialApp(
        debugShowCheckedModeBanner: false,
        theme: ThemeData.light(),
        darkTheme: ThemeData.dark(),
        title: 'Fleather - rich-text editor for Flutter',
        home: HomePage(),
      );
}

class HomePage extends StatelessWidget {
  const HomePage({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(elevation: 0, title: Text('Fleather Demo')),
      body: CustomScrollView(
        slivers: [
          SliverToBoxAdapter(child: Text('Header')),
          TextEditorX(),
          SliverToBoxAdapter(child: Text('Footer')),
        ],
      ),
    );
  }
}

class TextEditorX extends StatefulWidget {
  const TextEditorX({super.key});

  @override
  State<TextEditorX> createState() => _TextEditorXState();
}

class _TextEditorXState extends State<TextEditorX> {
  late ParchmentDocument document;
  late FleatherController _controller;

  @override
  void initState() {
    document = ParchmentDocument()..insert(0, 'Text');
    _controller = FleatherController(document);
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return SliverList(
      delegate: SliverChildListDelegate([
        FleatherToolbar.basic(controller: _controller),
        FleatherEditor(
          controller: _controller,
          padding: EdgeInsets.only(
            left: 16,
            right: 16,
            bottom: MediaQuery.of(context).padding.bottom,
          ),
          maxContentWidth: 800.0,
          maxHeight: 300.0,
        ),
      ]),
    );
  }
}
```

**Comment by Patrick386** [comment]
> I was able to use your widget in a `CustomScrollView` and it works fine. From the logs you've provided I think your issue is not related to `TextEditorX` widget. @Patrick386
> 
> ```dart
> import 'package:fleather/fleather.dart';
> import 'package:flutter/material.dart';
> 
> void main() {
>   runApp(const FleatherApp());
> }
> 
> class FleatherApp extends StatelessWidget {
>   const FleatherApp({Key? key}) : super(key: key);
> 
>   @override
>   Widget build(BuildContext context) => MaterialApp(
>         debugShowCheckedModeBanner: false,
>         theme: ThemeData.light(),
>         darkTheme: ThemeData.dark(),
>         title: 'Fleather - rich-text editor for Flutter',
>         home: HomePage(),
>       );
> }
> 
> class HomePage extends StatelessWidget {
>   const HomePage({Key? key}) : super(key: key);
> 
>   @override
>   Widget build(BuildContext context) {
>     return Scaffold(
>       appBar: AppBar(elevation: 0, title: Text('Fleather Demo')),
>       body: CustomScrollView(
>         slivers: [
>           SliverToBoxAdapter(child: Text('Header')),
>           TextEditorX(),
>           SliverToBoxAdapter(child: Text('Footer')),
>         ],
>       ),
>     );
>   }
> }
> 
> class TextEditorX extends StatefulWidget {
>   const TextEditorX({super.key});
> 
>   @override
>   State<TextEditorX> createState() => _TextEditorXState();
> }
> 
> class _TextEditorXState extends State<TextEditorX> {
>   late ParchmentDocument document;
>   late FleatherController _controller;
> 
>   @override
>   void initState() {
>     document = ParchmentDocument()..insert(0, 'Text');
>     _controller = FleatherController(document);
>     super.initState();
>   }
> 
>   @override
>   Widget build(BuildContext context) {
>     return SliverList(
>       delegate: SliverChildListDelegate([
>         FleatherToolbar.basic(controller: _controller),
>         FleatherEditor(
>           controller: _controller,
>           padding: EdgeInsets.only(
>             left: 16,
>             right: 16,
>             bottom: MediaQuery.of(context).padding.bottom,
>           ),
>           maxContentWidth: 800.0,
>           maxHeight: 300.0,
>         ),
>       ]),
>     );
>   }
> }
> ```
This test page is a page navigated through GoRouter. 
I've tried deleting everything and testing, but the same error keeps occurring. 
Flutter can be quite unfriendly, especially in its core parts, when it comes to the web. 
I've been using Flutter for over 4 years, but I still haven't found a single editor built with basic widgets. 
I've tried to create one, but there are countless unknown errors, dependency issues, and especially severe problems with physical keyboards. I'm waiting for all the frustrations to be resolved in WASM. 
Anyway, thank you for your quick and helpful response.

**Comment by Amir-P** [comment]
No problem! Will you able to run the sample code I've sent? Was it ok? @Patrick386

**Comment by Patrick386** [comment]
> No problem! Were you able to run the sample code I've sent? Was it ok? @Patrick386

I found the issue. The sample code is fine. The problem arose when using the sample JSON as it is. I retested it with the code you provided, and there is no issue. However, when entering Korean characters, it follows the cursor like in the video. Is there a way to resolve this issue?

https://github.com/fleather-editor/fleather/assets/49754877/46b56e1f-3f26-4826-8a5d-4f7df452dfd3
